lib/rpc/validation.js

```diff
@@ -0,0 +1,68 @@
+const { INVALID_PARAMS } = require('./error-code')
+
+module.exports = {
+  /**
+   * middleware for parameters validation
+   * @param {Function} method            function to add middleware
+   * @param {number} requiredParamsCount required parameters count
+   * @param {Function[]} validators      array of validator
+   */
+  middleware (method, requiredParamsCount, validators) {
+    return function (params, cb) {
+      if (params.length < requiredParamsCount) {
```

Comment:
Wonder if this should be a ``!==`` comparison and a bit more general error msg, but we can leave this for now. Another possibility would also to make an extra check for ``>``, this is maybe even more user-friendly for the caller.

Reply:
I did this because want to align with go-ethereum behaviors, but you are right we can handle this better than geth :D